events-api/src/shared/decorators/get-user-id.decorator.ts

```diff
@@ -0,0 +1,8 @@
+import { createParamDecorator, ExecutionContext } from '@nestjs/common';
+
+export const UserId = createParamDecorator(
+  (data: unknown, ctx: ExecutionContext) => {
+    const request = ctx.switchToHttp().getRequest();
+    return request.user?.userId;
```

Comment:
What will happen if user is empty?

Reply:
It will return undefined.

Now I fix it, I added checking which can return UnauthorizedException